custom/icds_reports/utils/aggregation_helpers/ccs_record_monthly.py

```diff
@@ -52,62 +52,139 @@ def tablename(self):
-            ('has_aadhar_id', 'ucr.has_aadhar_id'),
+            ('trimester_2', 'CASE WHEN {}=2 THEN 1 ELSE 0 END'.format(trimester)),
+            ('trimester_3', 'CASE WHEN {}=3 THEN 1 ELSE 0 END'.format(trimester)),
+            ('counsel_immediate_bf', 'CASE WHEN {}=3 THEN COALESCE(agg_bp.immediate_breastfeeding,0) '
+                                     'ELSE 0 END'.format(trimester)),
+            ('counsel_bp_vid', 'CASE WHEN {}=3 THEN COALESCE(agg_bp.play_birth_preparedness_vid,0) '
+                               'ELSE 0 END'.format(trimester)),
+            ('counsel_preparation', 'CASE WHEN {}=3 THEN COALESCE(agg_bp.counsel_preparation,0) '
+                                    'ELSE 0 END'.format(trimester)),
+            ('counsel_fp_vid', 'CASE WHEN {}=3 THEN COALESCE(agg_bp.play_family_planning_vid,0) '
+                               'ELSE 0 END'.format(trimester)),
+            ('counsel_immediate_conception', 'CASE WHEN {}=3 THEN COALESCE(agg_bp.conceive,0) '
+                                             'ELSE 0 END'.format(trimester)),
+            ('counsel_accessible_postpartum_fp', 'CASE WHEN {}=3 THEN COALESCE(agg_bp.counsel_accessible_ppfp,0) '
+                                                 'ELSE 0 END'.format(trimester)),
+            ('bp1_complete', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(b1_complete)),
+            ('bp2_complete', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(b2_complete)),
+            ('bp3_complete', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(b3_complete)),
+            ('pnc_complete', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(pnc_complete)),
+            ('postnatal', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(postnatal)),
+            ('has_aadhar_id', "CASE WHEN person_cases.aadhar_date < {} THEN"
+                              "  1 ELSE 0 END".format(end_month_string)),
             ('counsel_fp_methods', 'NULL'),
-            ('pregnant', 'ucr.pregnant'),
-            ('pregnant_all', 'ucr.pregnant_all'),
-            ('lactating', 'ucr.lactating'),
-            ('lactating_all', 'ucr.lactating_all'),
-            ('institutional_delivery_in_month', 'ucr.institutional_delivery_in_month'),
-            ('add', 'ucr.add'),
-            ('caste', 'ucr.caste'),
-            ('disabled', 'ucr.disabled'),
-            ('minority', 'ucr.minority'),
-            ('resident', 'ucr.resident'),
-            ('valid_in_month', 'ucr.valid_in_month'),
+            ('pregnant', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(pregnant_to_consider)),
+            ('pregnant_all', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(pregnant_all)),
+            ('lactating', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(lactating)),
+            ('lactating_all', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(lactating_all)),
+            ('institutional_delivery_in_month', 'CASE WHEN agg_delivery.where_born=3 AND {} THEN'
+                                                ' 1 ELSE 0 END'.format(delivered_in_month)),
+            ('add', 'case_list.add'),
```

Comment:
I think you only want to add this if `add` is beofre month end

Reply:
`delivered_in_month` ensured that.

Reply:
sorry, I somehow saw this comment was applied on 180 line. resolving this